### PR TITLE
S5 search bar

### PR DESCRIPTION
**Pull request by koma2323**
Added a SearchBar composable and added it in Events and Explore. With this it is now possible to filter the events displayed on the map by their title (the search function hasn't been implemented yet in Events since the events displayed in that tab are hard coded).

Capture d'écran 2024-04-21 141233
Capture d'écran 2024-04-21 141516

### CODE REVIEW

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/Events.kt [hunk 1]
```diff
@@ -13,19 +13,29 @@ import androidx.compose.foundation.layout.fillMaxSize
...
+import androidx.compose.foundation.layout.offset
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.layout.size
 import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
 import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
 import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Search
```
Comment (albertfares) [inline app/src/main/java/com/github/se/gomeet/ui/mainscreens/Events.kt:25]: use material3 instead

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/Events.kt
Comment (albertfares) [inline app/src/main/java/com/github/se/gomeet/ui/mainscreens/Events.kt:1]: The search bar UI looks good in general but there is an issue when we write some text: long letters (eg. 'j') don't fully show in the search bar. And dark mode makes the "Search" text and some icons way to light.

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/Explore.kt
Comment (albertfares) [inline app/src/main/java/com/github/se/gomeet/ui/mainscreens/Explore.kt:1]: Logic and UI look good. Same issues with long letters and dark mode

**Review by albertfares — CHANGES_REQUESTED** [review]
The search bar looks good in general and works well in the Explore screen. Some UI bugs still need to be fixed.

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/Events.kt (hunk 1 above)
Comment (koma2323) [reply app/src/main/java/com/github/se/gomeet/ui/mainscreens/Events.kt:26]: Capture d'écran 2024-04-21 131557
I searched on google but I can't find any material3 library for icons

Comment (albertfares) [reply app/src/main/java/com/github/se/gomeet/ui/mainscreens/Events.kt:26]: my bad, you're right

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/Events.kt
Comment (albertfares) [inline app/src/main/java/com/github/se/gomeet/ui/mainscreens/Events.kt:1]: the UI bug was fixed. thank you

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/Explore.kt
Comment (albertfares) [inline app/src/main/java/com/github/se/gomeet/ui/mainscreens/Explore.kt:1]: the UI bug was fixed. thank you

**Review by albertfares — APPROVED** [review]
The search bar looks good and working well now. The minor UI bugs were fixed.

**Pull request merged**